sinks/cortex/cortex_test.go

```diff
@@ -107,13 +108,56 @@ func TestChunkedWrites(t *testing.T) {
 	assert.Equal(t, 4, len(server.History()))
 }
 
+func TestMonotonicCounters(t *testing.T) {
+	// Listen for prometheus writes
+	server := NewTestServer(t)
+	defer server.Close()
+
+	// Set up a sink
+	sink, err := NewCortexMetricSink(server.URL, 30*time.Second, "", logrus.NewEntry(logrus.New()), "test", map[string]string{}, map[string]string{}, nil, 15, true)
+	assert.NoError(t, err)
+	assert.NoError(t, sink.Start(trace.DefaultClient))
+
+	// to_be_monotonic_counter.json contains 2 counters with the same name but different tags
+	// we want to be sure the hashing takes that into account
+	jsInput, err := ioutil.ReadFile("testdata/monotonic_counters.json")
+	assert.NoError(t, err)
+	var metrics []samplers.InterMetric
+	assert.NoError(t, json.Unmarshal(jsInput, &metrics))
+
+	_, err = sink.Flush(context.Background(), metrics)
+	assert.NoError(t, err)
+
+	_, err = sink.Flush(context.Background(), metrics)
+	assert.NoError(t, err)
+
+	expectedVals := map[string]float64{
+		"bar": 200,
+		"baz": 300,
+		"taz": 100,
+	}
+
+	matchesDone := 0
+	for _, data := range server.history[1].data.Timeseries {
+		fmt.Printf("%v", data)
+		for _, label := range data.Labels {
+			if label.Name == "foo" {
+				matchesDone++
+				assert.Equal(t, expectedVals[label.GetValue()], data.Samples[0].GetValue())
+			}
+		}
```

Comment:
It might be more readable to unroll this and use `assert.Contains(...)`
https://pkg.go.dev/github.com/stretchr/testify/assert#Contains